core/modules/path/path.admin.inc

```diff
@@ -223,18 +250,25 @@ function path_admin_form_validate($form, &$form_state) {
   // languages.
   $langcode = isset($form_state['values']['langcode']) ? $form_state['values']['langcode'] : LANGUAGE_NONE;
 
-  $has_alias = db_query("SELECT COUNT(alias) FROM {url_alias} WHERE pid <> :pid AND alias = :alias AND langcode = :langcode", array(
-      ':pid' => $pid,
-      ':alias' => $alias,
-      ':langcode' => $langcode,
-    ))
-    ->fetchField();
+  $query = db_select('url_alias', 'ua')
+    ->fields('ua', array('langcode'))
+    ->condition('pid', $pid, '<>')
+    ->condition('alias', $alias);
+  if ($langcode != LANGUAGE_NONE) {
+    $query->condition('langcode', array($langcode, LANGUAGE_NONE), 'IN');
+  }
+  $existing = $query->execute()->fetchCol();
 
-  if ($has_alias) {
-    form_set_error('alias', t('The alias %alias is already in use in this language.', array('%alias' => $alias)));
+  if (count($existing)) {
```

Comment:
Do we know that `$existing` will always be an array? I can't recall if `fetchCol()` will do that.

Reply:
According to [database.inc](https://github.com/backdrop/backdrop/blob/1.x/core/includes/database/database.inc#L2288) it's always an array.

Testing with an empty result set confirms that.